drivers/video/ov2640.c

```diff
@@ -152,17 +152,114 @@ LOG_MODULE_REGISTER(video_ov2640, CONFIG_VIDEO_LOG_LEVEL);
 #define REG32_UXGA          0x36
 
 /* Configuration arrays */
-#define SVGA_HSIZE     (800)
-#define SVGA_VSIZE     (600)
-
-#define UXGA_HSIZE     (1600)
-#define UXGA_VSIZE     (1200)
+#define SVGA_HSIZE (800UL)
+#define SVGA_VSIZE (600UL)
+
+#define UXGA_HSIZE (1600UL)
+#define UXGA_VSIZE (1200UL)
+
+#define VAL_SET(x, mask, rshift, lshift) ((((x) >> rshift) & mask) << lshift)
```

Comment:
Is that possible to use `FIELD_PREP()`? This would also break the chain from GPL code to Zephyr doing so.